src/ai/defaultai.cc

```diff
@@ -3330,18 +3330,34 @@ void DefaultAI::check_flag_distances(const Time& gametime) {
 // Dealing with diplomacy actions
 void DefaultAI::diplomacy_actions(const Time& gametime) {
 
+	// don't do any diplomacy for the first 10 + x minutes to avoid click races for allies
+	if (gametime < Time((10 + RNG::static_rand(10)) * 60 * 1000)) {
+		return;
+	}
```

Comment:
Shouldn't the initial delay be moved to the AI init function where diplomacy actions are first scheduled instead of checking every time?

Reply:
good idea